### PR TITLE
Fixes type definition for @types/fontkit to support FontCollection

### PR DESCRIPTION
**Pull request by talhabalaj**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/foliojs/fontkit#readme
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/fontkit/fontkit-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ed58b350c5bed31d592bf65c939698066bb8634a
</p>

**Comment by talhabalaj** [comment]
Builds on work by [loveloki](https://github.com/loveloki)  and #67233 which never got merged.

File: types/fontkit/fontkit-tests.ts [hunk 1]
```diff
@@ -24,37 +30,50 @@ const f = fontkit.openSync("fonts/a-font.ttf");
...
+
+// ------------------
+// API: FontCollection
+// -----------------
+openV2.then(font => {
+    if (isCollection(font)) {
+        font.fonts; // $ExpectType Font[]
+        font.getFont("hhe") // $ExpectType Font | null
+    }
+})
```
Comment (talhabalaj) [inline types/fontkit/fontkit-tests.ts:79]: open returns either a Font or a FontCollection. It can be distinguished by `type`.

**Comment by talhabalaj** [comment]
@stepancar Can you please review this?

**Review by Anonymous4078 — APPROVED** [review]

**Comment by talhabalaj** [comment]
Ready to merge

**Pull request merged**